authorizations/authorization-ranger/src/main/java/org/apache/gravitino/authorization/ranger/RangerAuthorizationHadoopSQLPlugin.java

```diff
@@ -76,6 +85,328 @@ public Map<Privilege.Name, Set<AuthorizationPrivilege>> privilegesMappingRule()
+                if (policy
+                    .getName()
+                    .equals(
+                        AuthorizationSecurableObject.DOT_JOINER.join(oldAuthzMetaobject.names()))) {
+                  List<String> policyNames =
+                      Lists.newArrayList(
+                          AuthorizationSecurableObject.DOT_SPLITTER.splitToList(policyName));
+                  Preconditions.checkArgument(
+                      policyNames.size() >= oldAuthzMetaobject.names().size(),
+                      String.format("The policy name(%s) is invalid!", policyName));
+                  if (policyNames.get(index).equals(RangerHelper.RESOURCE_ALL)) {
+                    // Doesn't need to rename the policy `*`
+                    return;
+                  }
+                  policyNames.set(index, newAuthzMetaobject.names().get(index));
+                  policy.setName(AuthorizationSecurableObject.DOT_JOINER.join(policyNames));
+                }
+                // Update the policy resource name to new name
+                policy
+                    .getResources()
+                    .put(
+                        policyResourceDefinesRule().get(index),
+                        new RangerPolicy.RangerPolicyResource(
+                            newAuthzMetaobject.names().get(index)));
+
+                boolean alreadyExist =
+                    existNewPolicies.stream()
+                        .anyMatch(
+                            existNewPolicy ->
+                                existNewPolicy.getName().equals(policy.getName())
+                                    || existNewPolicy.getResources().equals(policy.getResources()));
+                if (alreadyExist) {
+                  LOG.warn(
+                      "The Ranger policy for the metadata object({}) already exists!",
+                      newAuthzMetaobject);
+                  return;
+                }
+
+                // Update the policy
+                rangerClient.updatePolicy(policy.getId(), policy);
+              } catch (RangerServiceException e) {
+                LOG.error("Failed to rename the policy {}!", policy);
+                throw new RuntimeException(e);
+              }
+            });
+  }
+
+  /**
+   * IF remove the SCHEMA, need to remove these the relevant policies, `{schema}`, `{schema}.*`,
+   * `{schema}.*.*` <br>
+   * IF remove the TABLE, need to remove these the relevant policies, `{schema}.*`, `{schema}.*.*`
+   * <br>
+   * IF remove the COLUMN, Only need to remove `{schema}.*.*` <br>
+   */
+  @Override
+  protected void doRemoveMetadataObject(AuthorizationMetadataObject authzMetadataObject) {
+    AuthorizationMetadataObject.Type type = authzMetadataObject.type();
+    if (type.equals(SCHEMA)) {
+      doRemoveSchemaMetadataObject(authzMetadataObject);
+    } else if (type.equals(TABLE)) {
+      doRemoveTableMetadataObject(authzMetadataObject);
```

Comment:
Could we use a common method instead of three methods? Why do need the `PATH` case?

Reply:
1. These three methods are different.
2. Remove `PATH` case.

Reply:
After reading code, the difference is the data structure instead of processing logic. Maybe we can use an array or map for difference and extract a common method.

Reply:
I refactor this code.